### ISSUE TITLE
Minimised feature detection not handled

### ISSUE THREAD

**Issue by curiousdannii** [amilajack/eslint-plugin-compat#508]
This plugin is fine with this code
```
if (window.ResizeObserver) {
    this.canvasResizeObserver = new ResizeObserver((entries) => this.oncanvasresize(entries));
}
```

But I run it on the final Terser minimised output of my project (so that I can ensure that the final output is compatible), and it raises errors for this minimised version.
```
window.ResizeObserver&&(this.canvasResizeObserver=new ResizeObserver((e=>this.oncanvasresize(e))))
```

Can the feature detection mode be update to support `&&`?

**Comment by byFish** [comment]
check source code, I find out that was weird logic.
before report error, plugin check code is inside if statement, but without check conditional expressions.
Is this a bug?
```js
// eslint-plugin-compat/src/helpers.ts

function isInsideIfStatement(context: Context) {
  return context.getAncestors().some((ancestor) => {
    return ancestor.type === "IfStatement";
  });
}

function checkNotInsideIfStatementAndReport(
  context: Context,
  handleFailingRule: HandleFailingRule,
  failingRule: AstMetadataApiWithTargetsResolver,
  node: ESLintNode
) {
  if (!isInsideIfStatement(context)) {
    handleFailingRule(failingRule, node);
  }
}
```
